crates/proto/src/tcp/tcp_client_stream.rs

```diff
@@ -42,7 +42,7 @@ impl<S: Connect + 'static + Send> TcpClientStream<S> {
         name_server: SocketAddr,
     ) -> (
         TcpClientConnect<S::Transport>,
-        Box<dyn DnsStreamHandle + Send>,
+        Box<dyn DnsStreamHandle + 'static + Send>,
```

Comment:
can this be an impl returned?

Reply:
there are a bunch of downstream locations where this will impact usability. This can be revisited if ever opaque types can be named in the future.